### PR TITLE
AM-2764 SSCS_WA - CASE role validation

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-2764

### Change description ###
Add CASE role validation drool rules in RAS application for STAFF And JUDICIARY

(Using AM-2614 as base branch as there is some crossover in changes to JSON files)

### CODE REVIEW

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl [hunk 1]
```diff
@@ -65,3 +66,69 @@ then
...
+
+
+/*
+ * SSCS "allocated-tribunal-caseworker" case validation rule.
+ */
+rule "sscs_allocated_tribunal_caseworker_create_case_roles"
+when
+    $ca: CaseAllocatorApproval(
+                 roleAssignment.status == Status.CREATE_REQUESTED,
+                 roleAssignment.attributes["jurisdiction"].asText() == "SSCS",
```
Comment (abul-syed) [inline src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:78]:
```suggestion
                 roleAssignment.attributes["jurisdiction"].asText() == "SSCS",
                 roleAssignment.attributes["caseType"].asText() == "Benefit",
```

Noticed you left out the caseType check

Comment (abul-syed) [reply src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:78]: same comment for the 2 rules below

**Review by abul-syed — COMMENTED** [review]
just a few changes, then good to go

**Review by abul-syed — DISMISSED** [review]

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl [hunk 2]
```diff
@@ -42,6 +42,7 @@ then
     logMsg("Rule : sscs_tribunal_caseworker_create_case_roles");
```
Comment (mattnayler) [inline src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:42]: Is this rule, `sscs_tribunal_caseworker_create_case_roles`, to be removed.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java [hunk 3]
```diff
@@ -207,14 +235,23 @@ private void verifyGrantOrRejectAccessFor_CaseRole(String jurisdiction, String c
...
+        "SSCS,Benefit,tribunal-member-1",
+        "SSCS,Benefit,tribunal-member-2",
+        "SSCS,Benefit,tribunal-member-3",
+        "SSCS,Benefit,appraiser-1",
+        "SSCS,Benefit,appraiser-2",
         "SSCS,Benefit,panel-doctor",
         "SSCS,Benefit,panel-disability",
         "SSCS,Benefit,panel-financial",
         "SSCS,Benefit,panel-appraisal-judge",
         "SSCS,Benefit,panel-appraisal-medical",
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:247]: Are these still required:
        "SSCS,Benefit,panel-doctor",
        "SSCS,Benefit,panel-disability",
        "SSCS,Benefit,panel-financial",
        "SSCS,Benefit,panel-appraisal-judge",
        "SSCS,Benefit,panel-appraisal-medical",

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java [hunk 4]
```diff
@@ -207,14 +235,23 @@ private void verifyGrantOrRejectAccessFor_CaseRole(String jurisdiction, String c
...
         "SSCS,Benefit,interloc-judge",
+        "SSCS,Benefit,post-hearing-judge",
         "SSCS,Benefit,case-allocator",
         "SSCS,Benefit,registrar",
+        "SSCS,Benefit,allocated-tribunal-caseworker",
+        "SSCS,Benefit,allocated-admin-caseworker",
+        "SSCS,Benefit,allocated-ctsc-caseworker",
         "SSCS,Benefit,tribunal-caseworker",
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:255]: Is this still required:
   "SSCS,Benefit,tribunal-caseworker"

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl (hunk 2 above)
Comment (mattnayler) [reply src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:42]: NB: There is also test in `CaseRolesDroolsTest.java` for this.

**Review by abul-syed — APPROVED** [review]

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl (hunk 2 above)
Comment (mattnayler) [reply src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl]: Resolved in 83bf196b8348c5f9c90a2b71701377756c7b5f80.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java (hunk 3 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java]: Resolved in 83bf196b8348c5f9c90a2b71701377756c7b5f80.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java (hunk 4 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java]: Resolved in 83bf196b8348c5f9c90a2b71701377756c7b5f80.

**Review by mattnayler — APPROVED** [review]
Re-reviewed after 32152415476cd5cb15811d5c768220020fe1a3d9.

**Pull request merged**